app/models/post.rb

```diff
@@ -15,9 +16,17 @@ def five_recent_comments
     comments.order(created_at: :desc).limit(5)
   end
 
+  def liked_by?(user)
+    likes.exists?(author: user)
+  end
+
   private
 
-  def update_posts_counter
-    author.increment!(:posts_counter)
+  def update_posts_counter(increment: true)
+    if increment
+      author.increment!(:posts_counter)
+    else
+      author.decrement!(:posts_counter)
+    end
```

Comment:
- [ ] Kindly take note that I was unable to delete post as an `Admin` and `Authorised User`. I got the error message screenshot below 👇 

<img width="373" alt="Okota flyer-modified (3)" src="https://github.com/Timbar09/Blog_App/assets/103386183/0aed6ef7-2faf-43cd-b6b5-255bfc4dc7af">

Reply:
Fixed.